src/main/java/edu/indiana/dlib/amppd/service/impl/GalaxyApiServiceImpl.java

```diff
@@ -34,39 +32,77 @@
 	/**
 	 * @see edu.indiana.dlib.amppd.service.GalaxyApiService.getApiKey()
 	 */
-	public String getApiKey() {
-		/* Galaxy api key can be obtained via a GET request with baseauth, for ex.
-		request: curl –user zipzap@foo.com:password http://localhost:8080/api/authenticate/baseauth
-		response: {“api_key”: “baa4d6e3a156d3033f05736255f195f9” }
+	public GalaxyUser getCurrentUser() {
+		/* TODO
+		 * Below is a stub for the real implementation: for now we will return a new instance of the amppd master user as the current user;
+		 * once we set up user management and access control in AMP, we shall retrieve this info from the current user session.  
 		 */
-		/* TODO 
-		 * For now we use a master amppd user for all API requests; 
-		 * in the future we shall retrieve user/password from current AMP session, assuming the user already has a Galaxy account set up with same credentials as in AMP.
-		 */ 
-		RestTemplate authRestTemplate = new RestTemplate(new HttpComponentsClientHttpRequestFactoryBasicAuth(new HttpHost(config.getHost(), config.getPort(), "http")));
-		authRestTemplate.getInterceptors().add(new BasicAuthenticationInterceptor(config.getUsername(), config.getPassword()));
-		String url = config.getApiUrl() + "/authenticate/baseauth";
-		GalaxyApiKey key = authRestTemplate.getForObject(url, GalaxyApiKey.class);				
-		return key.getApi_key();
+		GalaxyUser user = new GalaxyUser();
+		user.setUsername(config.getUsername());
+		user.setPassword(config.getPassword());
+		return user;		
 	}
-		
-    /**
-     * @see edu.indiana.dlib.amppd.service.GalaxyApiService.getWorkflowUrl()
-     */
-    public String getWorkflowUrl() {
-    	return config.getApiUrl() + "/workflows";
-    }
-
+	
 	/**
-	 * @see edu.indiana.dlib.amppd.service.GalaxyApiService.getWorkflows()
+	 * @see edu.indiana.dlib.amppd.service.GalaxyApiService.getInstance()
 	 */
-	@GetMapping("/workflows")
-	public GalaxyWorkflow[] getWorkflows() {		
-		String url = getWorkflowUrl() + "?key=" + getApiKey();
-		GalaxyWorkflow[] workflows = restTemplate.getForObject(url, GalaxyWorkflow[].class);
-		log.info("Current workflows in Galaxy: " + workflows);
-		return workflows;
+	public GalaxyInstance getInstance() {
```

Comment:
@yingfeng-iu can we implement exception handling, incase galaxy returns an error?